crates/accelerate/src/commutation_checker.rs

```diff
@@ -0,0 +1,770 @@
+    }
+
+    if is_commutation_skipped(op1, params1) || is_commutation_skipped(op2, params2) {
+        return Some(false);
+    }
+
+    None
+}
+
+fn get_matrix(py: Python, operation: &OperationRef, params: &[Param]) -> Option<Array2<Complex64>> {
+    match operation.matrix(params) {
+        Some(matrix) => Some(matrix),
+        None => match operation {
+            PyGateType(gate) => matrix_via_operator(py, &gate.gate),
+            PyOperationType(op) => matrix_via_operator(py, &op.operation),
+            _ => None,
+        },
+    }
+}
+
+fn matrix_via_operator(py: Python, py_obj: &PyObject) -> Option<Array2<Complex64>> {
+    Some(
+        QI_OPERATOR
+            .get_bound(py)
+            .call1((py_obj,))
+            .ok()?
+            .getattr(intern!(py, "data"))
+            .ok()?
+            .extract::<PyReadonlyArray2<Complex64>>()
+            .ok()?
+            .as_array()
+            .to_owned(),
+    )
+}
+
+fn is_commutation_skipped<T>(op: &T, params: &[Param]) -> bool
+where
+    T: Operation,
+{
+    op.directive()
+        || SKIPPED_NAMES.contains(&op.name())
+        || params
+            .iter()
+            .any(|x| matches!(x, Param::ParameterExpression(_)))
+}
+
+fn get_relative_placement(
+    first_qargs: &[Qubit],
+    second_qargs: &[Qubit],
+) -> SmallVec<[Option<Qubit>; 2]> {
+    let qubits_g2: HashMap<_, _> = second_qargs
+        .iter()
+        .enumerate()
+        .map(|(i_g1, q_g1)| (q_g1, Qubit(i_g1 as u32)))
+        .collect();
+
+    first_qargs
+        .iter()
+        .map(|q_g0| qubits_g2.get(q_g0).copied())
+        .collect()
+}
+
+#[derive(Clone, Debug)]
+#[pyclass]
+pub struct CommutationLibrary {
+    pub library: Option<HashMap<(String, String), CommutationLibraryEntry>>,
+}
+
+impl CommutationLibrary {
+    fn check_commutation_entries(
+        &self,
+        first_op: &OperationRef,
+        first_qargs: &[Qubit],
+        second_op: &OperationRef,
+        second_qargs: &[Qubit],
+    ) -> Option<bool> {
+        if let Some(library) = &self.library {
+            match library.get(&(first_op.name().to_string(), second_op.name().to_string())) {
+                Some(CommutationLibraryEntry::Commutes(b)) => Some(*b),
+                Some(CommutationLibraryEntry::QubitMapping(qm)) => qm
+                    .get(&get_relative_placement(first_qargs, second_qargs))
+                    .copied(),
+                _ => None,
+            }
+        } else {
+            None
+        }
+    }
+}
+
+#[pymethods]
+impl CommutationLibrary {
+    #[new]
+    fn new(py_any: Option<Bound<PyAny>>) -> Self {
+        match py_any {
+            Some(pyob) => CommutationLibrary {
+                library: pyob
+                    .extract::<Option<HashMap<(String, String), CommutationLibraryEntry>>>()
+                    .unwrap(),
+            },
+            None => CommutationLibrary {
+                library: Some(HashMap::new()),
+            },
+        }
+    }
+}
```

Comment:
Instead of extracting the `Option` you could just extract the `HashMap` and default to `None` when anything invalid is passed. For this you can use `.ok()`:
```suggestion
    fn new(py_any: Option<Bound<PyAny>>) -> Self {
        match py_any {
            Some(pyob) => CommutationLibrary {
                library: pyob
                    .extract::<HashMap<(String, String), CommutationLibraryEntry>>()
                    .ok(),
            },
            None => CommutationLibrary {
                library: Some(HashMap::new()),
            },
        }
    }
}
```
If the goal instead was to get an empty `HashMap` when invalid. You can still modify it so that it unwraps things without panicking.

Reply:
Done in 12d0a05